src/wallet/load.cpp

```diff
@@ -77,27 +82,88 @@ bool LoadWallets(interfaces::Chain& chain, const std::vector<std::string>& walle
     return true;
 }
 
+static std::thread wbThread;
+static std::atomic_bool wbRun{false};
+static std::condition_variable_any wbWaiter;
+
 void StartWallets(CScheduler& scheduler)
 {
-    for (const std::shared_ptr<CWallet>& pwallet : GetWallets()) {
+    for (auto& pwallet : GetWallets()) {
         pwallet->postInitProcess();
     }
 
     // Schedule periodic wallet flushes and tx rebroadcasts
     scheduler.scheduleEvery(MaybeCompactWalletDB, 2000);
     scheduler.scheduleEvery(MaybeResendWalletTxs, 1000);
+
+    // Schedule periodic wallet backup
+    auto everyMinutes = gArgs.GetArg("-walletbackup", DEFAULT_WALLET_BACKUP_MINUTES);
+    if (everyMinutes > 0) {
+        auto backupWallet = []() {
+            for (auto& pwallet : GetWallets()) {
+                pwallet->BlockUntilSyncedToCurrentChain();
+
+                auto locked_chain = pwallet->chain().lock();
+                LOCK2(pwallet->cs_wallet, locked_chain->mutex());
+
+                auto walletFile = pwallet->GetLocation().GetFilePath();
+                if (fs::is_directory(walletFile)) {
+                    walletFile += "/wallet.dat";
+                }
+                if (!fs::exists(walletFile)) {
+                    continue;
+                }
+                auto now = GetTime();
+                auto prevBackup = strprintf("%s.auto.%d.bak1", walletFile, now);
+                if (fs::exists(prevBackup)) {
+                    fs::remove(prevBackup);
+                }
+                if (pwallet->BackupWallet(prevBackup)) {
+                    auto currBackup = strprintf("%s.auto.%d.bak", walletFile, now);
```

Comment:
This just renames `prevBackup` into `currBackup`, but is there a reason to do this, since prevbackup is a constructed name anyway with the timestamp. 

In order to remove the oldest file, we need to scan the fs, and remove the oldest one, leaving 2 intact.

Reply:
It's just tries to prevent someone else to inject a file to screw up the process.